### ISSUE TITLE
`exec` for copilot jobs

### ISSUE THREAD

**Issue by RaffaeleSgarro** [aws/copilot-cli#4756]
I have a bunch of programs deployed as "Scheulded jobs" and I can't ECS-Exec using Copilot CLI, even if [the documentation](https://aws.github.io/copilot-cli/docs/commands/svc-exec/) suggests it should be possible:

```
-n, --name string        Name of the service, job, or task group.
```
Actually when I execute the command with `--name MyScheduledJobName`  Copilot complains *No service found*. Did I misunderstood the documentation and is this a feature request?

**Comment by dannyrandall** [comment]
Hi @RaffaeleSgarro! `copilot svc exec` is only supported for _service_ types, not jobs. Currently supported services are Load Balanced Web, Backend, and Worker services.

I think Copilot could support `copilot job exec` in the future! Could you share a bit about your use case, and I can modify this issue to be a feature request? Do you have long running jobs that you're trying to debug? Are you waiting for a scheduled job to start and quickly trying to exec into it? Thanks! 😊

**Comment by RaffaeleSgarro** [comment]
Sure! We have some long-running cron jobs (lasting minutes to months, it depends on the job) deployed as "Scheduled Job" and deployed with copilot in order to have *serverless cron* and *serverless batch*. Some jobs are triggered `@daily` and others are `schedule: none` and actually run with `copilot job run`.

First of all: if there are better tools than Copilot to manage this kind of workload on AWS feel free to suggest them. Otherwise we generally like Copilot way of doing things and are happy with it, but it lacks lots of functionalities for "Scheduled Job's" that are only available for services and/or tasks (for example `copilot job stop`?). Anyway most of our programs are written in Java, which is a highly instrumentable platform that we can easily instrument on EC2 but somhow not in containers, for example:
- taking heap and thread dumps by ECS-exec'ing into the container
- inspect JVM crash files

In this case ECS-exec'in into the container is needed to investigate the filesystem to troubleshoot a bug, and to use local tools to take a thread dump.

**Comment by dannyrandall** [comment]
Awesome, thanks for the feedback! Makes sense to me. Also `job stop` sounds cool too! I updated the issue to be a feature request!😊

**Comment by RaffaeleSgarro** [comment]
@dannyrandall it may be easier than it sounds. We found a parameter `EnableExecuteCommand` in the [AWS Step Functions documentation](https://docs.aws.amazon.com/step-functions/latest/dg/connect-ecs.html) so adding it in `internal/pkg/template/templates/workloads/partials/cf/state-machine-definition.json.yml` should be enough to start the agent. Then the command `copilot job exec` works the same way as `svc exec` and just has to locate the task with the AWS SDK

**Comment by dannyrandall** [comment]
@RaffaeleSgarro Thanks for looking into it - I think you're right! You could use [YAML patch overrides](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) to modify the state machine with the `EnableExecuteCommand` parameter, though it's not perfect since the definition is a JSON string, and you'd have to lookup the ECS task and run exec yourself with the AWS CLI 😔.

If you'd like, we'd also love a PR to add the feature! I'd be happy to help point you in the right direction if you're interested.😊

**Comment by RaffaeleSgarro** [comment]
@dannyrandall yes I'd love to contribute. I already tried but am not a Go programmer and the repository doesn't contain a description of the development environment. I opened the project with Go Land (the Jetbrains IDE) and found the Makefile, but was not able to do nothing more than a static analysis of the code to understand what's going on. Do you have any internal resource about configuring the development environment, debugging (with breakpoints) and running tests? I'm on Windows

**Comment by dannyrandall** [comment]
Awesome!💛 Does the [CONTRIBUTING.md](https://github.com/aws/copilot-cli/blob/mainline/CONTRIBUTING.md) doc help answer any questions? GoLand is a great IDE for Go - it should have the ability to run any of the unit tests (files that end in `_test.go`). For debugging, I mostly use `make build` to build a local version of Copilot and test with that. For example, you could modify the job state machine definition, run `make build` and then use the local version of Copilot to test deploying a modified job with it (`./bin/local/copilot job deploy`)!
